packages/ui/src/modal-building-code-content/ModalBuildingCodeContent.tsx

```diff
@@ -0,0 +1,227 @@
+        ))) ??
+      null
+    );
+  };
+
+  const renderSections = (sections: SectionType[]) => {
+    return (
+      (!!sections &&
+        sections.map((data, index) => (
+          <div
+            key={index}
+            ref={(el) => {
+              sectionRefs.current[data.numberReference] = el;
+            }}
+            className={`${
+              highlightedSection === data.numberReference
+                ? "ui-ModalSide--SectionHighlighted"
+                : ""
+            }`}
+          >
+            {/* Designs do not show this high level of headings. */}
+            {data.subsections && renderSubSections(data.subsections)}
+          </div>
+        ))) ??
+      null
+    );
+  };
+
+  const renderSubSections = (subSections: SubsectionType[]) => {
+    return (
+      (!!subSections &&
+        subSections.map((data, index) => (
+          <article
+            key={index}
+            ref={(el) => {
+              sectionRefs.current[data.numberReference] = el;
+            }}
+            className={`${
+              highlightedSection === data.numberReference
+                ? "ui-ModalSide--SectionHighlighted"
+                : ""
+            }`}
+          >
+            <header className="ui-ModalSide--SectionHeaderLine">
+              <Heading className="ui-ModalSide--SectionHeaderLarge">
+                <span className="ui-ModalSide--SectionNumberLarge">
+                  {stripReferencePrefix(data.numberReference)}
+                </span>
+                {data.title}
+              </Heading>
+            </header>
+            {data.articles && renderArticles(data.articles)}
+          </article>
+        ))) ??
+      null
+    );
+  };
+
+  const renderArticles = (articles: ArticleType[]) => {
+    return (
+      (!!articles &&
+        articles.map((data, index) => (
+          <article
+            key={index}
+            ref={(el) => {
+              sectionRefs.current[data.numberReference] = el;
+            }}
+            className={`${
+              highlightedSection === data.numberReference
+                ? "ui-ModalSide--SectionHighlighted"
+                : ""
+            }`}
+          >
+            <header className="ui-ModalSide--SectionHeaderLine">
+              <Heading className="ui-ModalSide--SectionHeader">
+                <span className="ui-ModalSide--SectionNumber">
+                  {stripReferencePrefix(data.numberReference)}
+                </span>
+                {data.title}
+              </Heading>
+            </header>
+            {data.sentences && renderSentences(data.sentences)}
+          </article>
+        ))) ??
+      null
+    );
+  };
+
+  const renderSentences = (sentences: SentenceType[]) => {
+    return (
+      <div className="ui-ModalBuildingCodeContent--SectionLine">
+        <div className="ui-ModalSide--SectionContent">
+          <ol type="1">
+            {sentences.map((sentence, index) => (
+              <section
```

Comment:
I'm not sure we can have the `<section>` elements as children of the `<ol>`'s semantically. It looks like it might take different CSS solutions to get the highlighting style to work correctly then. I would also think we don't want these to be `≤section>` elements. `<ol>` and their `<li>`'s already have an appropriate amount of semantic purpose here. This might be good to add to the polish ticket.